packages/jest-cli/src/__tests__/test_sequencer.test.js

```diff
@@ -11,18 +11,47 @@ import path from 'path';
 import TestSequencer from '../TestSequencer';
 
 jest.mock('fs');
+jest.mock(
+  'jest-resolve-dependencies',
+  () =>
+    class {
+      constructor() {
+        this.resolveRecursive = jest
+          .fn()
+          .mockImplementation((path, options) => {
+            expect(options.includeCoreModules).toBe(true);
```

Comment:
having an assertion here is pretty weird, could you do have the mock outside of this closure, then `expect(resolveRecursiveMock).toHaveBeenCalledWith(expect.any(string), {includeCoreModules: true})`, making sure to reset the calls between tests?

It's a bit more code, but it decouples the mock functions from assertions about the mock call